packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -32,6 +34,51 @@ export interface TableProps {
     streamSpecification?: StreamViewType;
 }
 
+export interface DynamoDBAutoScalingProps {
+    /* tslint:disable:max-line-length */
+    /**
+     * The minimum value that Application Auto Scaling can use to scale a target during a scaling activity.
+     * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-applicationautoscaling-scalabletarget.html#cfn-applicationautoscaling-scalabletarget-mincapacity
+     */
+    /* tslint:enable:max-line-length */
+    minimumCapacity: number;
+    /* tslint:disable:max-line-length */
+    /**
+     * The maximum value that Application Auto Scaling can use to scale a target during a scaling activity.
+     * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-applicationautoscaling-scalabletarget.html#cfn-applicationautoscaling-scalabletarget-maxcapacity
+     */
+    /* tslint:enable:max-line-length */
+    maximumCapacity: number;
+    /* tslint:disable:max-line-length */
+    /**
+     * The target value for the metric.
+     * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-applicationautoscaling-scalingpolicy-targettrackingscalingpolicyconfiguration.html#cfn-applicationautoscaling-scalingpolicy-targettrackingscalingpolicyconfiguration-targetvalue
+     */
+    /* tslint:enable:max-line-length */
+    scalingTargetValue: number;
+    /* tslint:disable:max-line-length */
+    /**
+     * The amount of time, in seconds, after a scale in activity completes before another scale in activity can start.
+     * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-applicationautoscaling-scalingpolicy-targettrackingscalingpolicyconfiguration.html#cfn-applicationautoscaling-scalingpolicy-targettrackingscalingpolicyconfiguration-scaleincooldown
+     */
+    /* tslint:enable:max-line-length */
+    scaleInCooldown: number;
+    /* tslint:disable:max-line-length */
+    /**
+     * The amount of time, in seconds, after a scale out activity completes before another scale out activity can start.
+     * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-applicationautoscaling-scalingpolicy-targettrackingscalingpolicyconfiguration.html#cfn-applicationautoscaling-scalingpolicy-targettrackingscalingpolicyconfiguration-scaleoutcooldown
+     */
+    /* tslint:enable:max-line-length */
+    scaleOutCooldown: number;
+    /* tslint:disable:max-line-length */
+    /**
+     * A name for the scaling policy.
+     * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-applicationautoscaling-scalingpolicy.html#cfn-applicationautoscaling-scalingpolicy-policyname
+     */
+    /* tslint:enable:max-line-length */
+    scalingPolicyName: string;
```

Comment:
Can we autogenerate a name if the user doesn't specify one?

Reply:
I'd like to discuss this one. There's places where CFN considers policy name to be optional and populates it without the user needing to know about it. Since it is mandatory here, I'm not sure if we should hide this from the user. That also means CDK will need to responsible for maintaining uniqueness with respect to the name.